docusaurus/src/components/RequestERD/EmailModal.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import styles from "./EmailModal.module.css";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import cn from "classnames";
+import { useDebouncedCallback } from "use-debounce";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+
+const CloseButton = ({ onClose }) => {
+  return (
+    <button onClick={onClose} className={styles.modalContentClose}>
+      <FontAwesomeIcon icon={faXmark} />
+    </button>
+  );
+};
+
+const ModalBody = ({ status, onSubmit }) => {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+
+  const debouncedValidation = useDebouncedCallback((value) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (emailRegex.test(value)) {
+      setEmailError("");
+    } else {
+      setEmailError("Please enter a valid email address");
+    }
+  }, 800);
+
+  const handleOnChange = (e) => {
+    const emailValue = e.target.value;
+    setEmail(emailValue);
+    debouncedValidation(emailValue);
+  };
+
+  if (status === "success") {
+    return (
+      <div>
+        <p> Request submitted successfully!</p>
+        <p> We will notify you when the ERD is ready</p>
+      </div>
+    );
+  }
+  if (status === "error") {
+    return (
+      <div>
+        <p> Error submitting request. Please try again later.</p>
+      </div>
+    );
+  }
+
+  return (
+    <form onSubmit={() => onSubmit(email)} className={styles.modalContentForm}>
+      <div className={styles.modalContentInputContainer}>
+        <input
+          className={cn(styles.modalContentInput, {
+            [styles.modalContentInputError]: emailError,
+          })}
+          type="email"
+          value={email}
+          autoComplete="email"
+          onChange={handleOnChange}
+          placeholder="Enter your email"
+          required
+          disabled={status === "loading"}
+        />
+        {emailError && <p className={styles.inputErrorMessage}>{emailError}</p>}
+      </div>
+      <button
+        type="submit"
+        onClick={(e) => {
+          e.preventDefault();
+          onSubmit(email);
+        }}
+        className={styles.modalContentButton}
+        disabled={status === "loading"}
+      >
+        {status === "loading" ? "Submitting..." : "Submit"}
+      </button>
+    </form>
+  );
+};
+
+export const EmailModal = ({ isOpen, onClose, sourceInfo }) => {
+  const [status, setStatus] = useState("");
+  const {
+    siteConfig: {
+      customFields: { requestErdApiUrl },
+    },
+  } = useDocusaurusContext();
+ 
+
+  if (!isOpen) return null;
+
+  const handleSubmit = async (email) => {
```

Comment:
I always get an error when submitting my email, even if my email is valid.

Reply:
I’m sorry, I should have explained how to test this in the PR.
This submission is sent to an api endpoint in the `airbyte-web-experiments` project, which is also hosted in vercel.
The [endpoint](https://github.com/airbytehq/airbyte-web-experiments/blob/935248635f1713d23cd2557fdcb41c9f3615a6eb/pages/api/request-erd/index.ts#L6) will take care of triggering the Retool workflow.
Since origins are different for both projects, I had to set CORS at the [middleware](https://github.com/airbytehq/airbyte-web-experiments/blob/935248635f1713d23cd2557fdcb41c9f3615a6eb/middleware.ts#L7) 
Anyway, I had an error on the env variables that make this possible.

If you test it in the [preview](https://airbyte-docs-git-leti-allow-request-erds-on-docs-airbyte-growth.vercel.app/integrations/sources/activecampaign) should work now.